http/template/assets.go

```diff
@@ -0,0 +1,69 @@
+package template
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"net/url"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/xy-planning-network/trails"
+	"github.com/xy-planning-network/trails/logger"
+)
+
+const (
+	assetsBase = "client/dist"
+)
+
+// AssetURI encloses the environment and filesystem so when called executing a template,
+// emits valid URI for client side static and bundled assets.
+//
+// The asset origin defaults to "/" when it is not configured and is always set with a trailing slash.
+func AssetURI(origin *url.URL, env trails.Environment, filesys fs.FS, l logger.Logger) func(string) string {
```

Comment:
Late to the party... but!

Does it make more sense to return an error alongside the path instead? The closure signature would be `func(string) (string, error)`, accordingly.

Then, if we're wanting to have some observability on this specific issue either:
1. error out and fall back to the error template in `ranger` where `resp.Html` will log the error, or,
2. throw another template function together that logs the error and throws it away, passing along the `string` the `AssetURI` closure returns.

I don't want to make this more complicated than it's worth with a purist critique. But, to me, including an initialized service in order to have observability functionality is not simple. It nudges the function towards more complexity along the lines of "AssertURI, toggles around the env to put together a path to an asset in the filesys that incorporates the origin, and logs when globbing fails to return a single asset".

Reply:
I don't think we have any template functions that return errors, so to a quick example: https://go.dev/play/p/3fptfpnruDd

Reply:
I do think there's a pretty decent argument for returning an error and having it fallback to the error template, particularly since finding multiple matches is likely going to mean a broken page that has no UI to speak of.  If we move in the direction of consuming the manifest, that could increase the surface area of new types of errors that occur, so having the ability to return errors of any kind an know that the error template is rendered when things go wrong sounds like a good idea.

Reply:
<img width="1490" alt="Screenshot 2023-09-27 at 4 56 26 PM" src="https://github.com/xy-planning-network/trails/assets/3856703/ad85953c-650f-43cd-9efd-bf04929967d0">

This feel very handy.